sql-parser/src/main/java/io/crate/sql/SqlFormatter.java

```diff
@@ -205,6 +206,41 @@ protected Void visitExplain(Explain node, Integer indent) {
             return null;
         }
 
+        @Override
+        public Void visitInsert(Insert<?> node, Integer indent) {
+            append(indent, "INSERT");
+            builder.append(' ');
+            append(indent, "INTO");
+            builder.append(' ');
+            node.table().accept(this, indent);
+            builder.append(' ');
+            var columns = node.columns().iterator();
+            if (columns.hasNext()) {
+                builder.append('(');
+                while (columns.hasNext()) {
+                    builder.append(columns.next());
+                    if (columns.hasNext()) {
+                        builder.append(", ");
+                    }
+                }
+                builder.append(')');
+            }
+            builder.append(' ');
+            node.insertSource().accept(this, indent);
```

Comment:
isn't `onConflict` still missing here?

Reply:
It is missing, although it is actually was in the https://github.com/crate/crate/blob/208553a80f988643b74ee3602ae5f46a7fcb32e9/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java#L1050
but not failing.